docs/best-practices/tech-debt.md

```diff
@@ -0,0 +1,218 @@
+out to all users, the flag no longer serves its purpose. Keeping the flag increases the chances of falling back to the
+old UI if the flag is accidentally turned off or if a system fails to integrate the SDK.
+
+It's therefore important to stay on top of your feature flags and remove them when they are no longer needed.
+
+## When to Remove a Flag
+
+The decision to remove a flag depends on what “type” of flag it is. DevCycle believes strongly in the philosophy
+described in the article **[Feature toggles (aka Feature Flags)](https://martinfowler.com/articles/feature-toggles.html) by Pete Hodgson**
+To summarize, there are four types of flags with varying expectations of their "lifespan" and when they should be removed:
+
+**Release**
+
+A flag which controls the release of a new feature. When the feature is fully released, the flag should be removed once
+the team is confident that the feature is stable and does not need to be turned off.
+
+**Experiment**
+
+A flag which controls an A/B test between multiple possible variations. Once the results of the experiment have 
+reached statistical significance and a winning variation can be determined, the flag should be replaced with the value
+of that winner.
+
+**Ops**
+
+These flags are used to control operation aspects of the system. They are expected to be long-lived and remain in code,
+and can be used for things such as turning off optional and non-essential features in the event of high load.
+
+**Permission**
+
+These flags are used to control access to a feature based on user permissions. They are expected to be long-lived and
+remain in code, and can be used for things such as turning off features for users who do not have the correct permissions
+to access them. 
+
+As you can see, Release and Experiment flags are expected to be short-lived and removed when no longer needed, whereas
+Ops and Permission flags are expected to be long-lived and remain in code.
+
+## Lifecycle of a Feature
+Features that are not expected to stay in code forever (ie. Release and Experiment Features) generally go through a
+similar set of lifecycle steps.
+
+For a release feature:
+1. Feature Variables are added to code and the Feature is set up in DevCycle
+2. Feature is enabled for particular groups of users throughout development
+3. Feature is enabled for all users when it is complete
+4. Feature is considered "complete" once it is fully released and stable, and the ability to turn it off is no longer
+necessary
+5. Feature's Variables are removed from code
+6. Once it can be verified that the Variables are no longer used in any running code, the Feature on DevCycle can be 
+archived
+
+For an experiment feature:
+1. Feature Variables are added to code and the Feature is set up in DevCycle
+2. Feature is enabled for a subset of users that are part of the experiment
+3. Feature is considered "complete" once it has reached statistical significance, and the winning variation can be determined.
```

Comment:
this is the only line with a period at the end, recommend removing to keep format the same